tools/ninja/patches/100-make_jobserver_support.patch

```diff
@@ -363,104 +294,67 @@ Fixes https://github.com/ninja-build/ninja/issues/1139
 +// See the License for the specific language governing permissions and
 +// limitations under the License.
 +
-+#include "tokenpool-gnu-make.h"
++#include "jobserver.h"
 +
-+#include <stdlib.h>
-+#include <stdio.h>
-+#include <string.h>
++#include <fcntl.h>
++#include <unistd.h>
 +
-+#include "line_printer.h"
++#include <cassert>
++#include <cstring>
 +
-+// TokenPool implementation for GNU make jobserver - common bits
-+// every instance owns an implicit token -> available_ == 1
-+GNUmakeTokenPool::GNUmakeTokenPool() : available_(1), used_(0) {
-+}
++#include "util.h"
 +
-+GNUmakeTokenPool::~GNUmakeTokenPool() {
-+}
++void Jobserver::Init() {
++  assert(fd_ < 0);
 +
-+bool GNUmakeTokenPool::Setup(bool ignore,
-+                             bool verbose,
-+                             double& max_load_average) {
-+  const char* value = GetEnv("MAKEFLAGS");
-+  if (!value)
-+    return false;
++  if (!ParseJobserverAuth("fifo")) {
```

Comment:
Martin's implementation indeed only supports Make 4.4

Reply:
If the other versions of this patch also support older make versions I would prefer that. Even Ubuntu 24.10 is still using make 4.3 same for Debian sid.